website_snippet_calendar/static/src/js/website_snippet_calendar.js

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2017 Onestein
+* License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('website_snippet_calendar', function(require) {
+'use strict';
+
+    var core = require('web.core'),
+        time = require('web.time'),
+        base = require('web_editor.base'),
+        _t = core._t;
+
+    base.ready().done(function() {
+        //Reset
+        var $calendar = $('.s_calendar .calendar');
+        $calendar.html('');
+
+        //Colors
+        var all_filters = {};
+        var color_map = {};
+
+        var get_color = function(key) {
+            if (color_map[key]) {
+                return color_map[key];
+            }
+            var index = (((_.keys(color_map).length + 1) * 5) % 24) + 1;
+            color_map[key] = index;
+            return index;
+        }
+
+        //Get date format
+        var dateFormat = time.strftime_to_moment_format(_t.database.parameters.date_format);
+
+        var conversions = [['YYYY', 'yyyy'], ['YY', 'y'], ['DDDD', 'dddd'], ['DD', 'dd']];
+        _.each(conversions, function(conv) {
+            dateFormat = dateFormat.replace(conv[0], conv[1]);
+        });
+
+        $calendar.fullCalendar({
+            events: function(start, end, callback) {
+                $.ajax({
+                    url: '/calendar_snippet/get_events/' + Math.round(start.getTime() / 1000) + '/' + Math.round(end.getTime() / 1000),
+                    dataType: 'json',
+                    success: function(result) {
+                        //Setup colors
+                        all_filters[0] = get_color(result.contacts[0]);
+                        all_filters[-1] = get_color(-1);
+                        _.each(result.contacts, function (c) {
+                            if (!all_filters[c]) {
+                                all_filters[c] = get_color(c);
+                            }
+                        });
+
+                        //Mutate data for calendar
+                        var events = result.events;
+                        for(var i = 0; i < events.length; i++) {
```

Comment:
what about moving this to specific function? Something like `prepare_events` would ease customization.